pkg/resource/deploy/step_generator.go

```diff
@@ -233,6 +233,9 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, res
 		}
 	}
 
+	// register aliases with the provider registry
+	sg.deployment.providers.RegisterAliases(sg.aliased)
```

Comment:
This is overkill: it's re-registering every alias after each resource registration. I think that we can move this into the code above that does the usual alias processing and change it s.t. it only handles a single alias at a time.